src/Build.UnitTests/BackEnd/IntrinsicTask_Tests.cs

```diff
@@ -1988,63 +1988,6 @@ public void KeepWithItemReferenceOnNonmatchingMetadata()
             items.ElementAt(3).GetMetadataValue("d").ShouldBe("d");
         }
 
-        [Fact]
-        public void FailWithMatchingMultipleMetadata()
```

Comment:
Don't delete the tests, instead replace the exception assertion with an assertion on I2's contents

Reply:
Changed it to "RemoveWithMatchingMultipleMetdata." I did enable matching multiple.